portability-spi-cloud/src/main/java/org/dataportabilityproject/spi/cloud/storage/JobStore.java

```diff
@@ -15,9 +15,16 @@
  * back-end services.
  */
 public interface JobStore {
+  interface JobUpdateValidator {
+    /**
+     * Validation to do as part of an atomic update. Implementers should throw an
+     * {@code IllegalStateException} if the validation fails.
+     */
+    void validate(PortabilityJob previous, PortabilityJob updated);
```

Comment:
So clients can see it in the contract of the method signature, let's explicitly declare throws IllegalStateException or a custom ValidationException

Reply:
IllegalStateException is unchecked, and this is the same convention as Preconditions.checkState (i.e. that also documents @throws IllegalStateException but does not explicitly declare in method signature)

I think it's probably unrecoverable if validate fails, so should stick with an unchecked exception, depending on what we decide for #176. Can we revisit this with #176?

Reply:
Sure, I'm thinking more about defining explicitly how a client handles exceptions to the validation to avoid a mishmash of inconsistent exception handling.